src/libsyntax/parse/parser/path.rs

```diff
@@ -196,12 +197,12 @@ impl<'a> Parser<'a> {
                 let (args, constraints) =
                     self.parse_generic_args_with_leaning_angle_bracket_recovery(style, lo)?;
                 self.expect_gt()?;
-                let span = lo.to(self.prev_span);
+                let span = ident.span.to(self.prev_span);
```

Comment:
This does not seem correct; An `AngleBracketedArgs` denotes the stuff in `<...>`, not the whole `foo::<...>`. The code as-is is correct.

Reply:
Got overeager in my changes, I assumed that it was the same as the `Fn()` trait names.